addons/account/wizard/account_payment_register.py

```diff
@@ -115,7 +115,10 @@ def _get_batch_communication(self, batch_result):
         :param batch_result:    A batch returned by '_get_batches'.
         :return:                A string representing a communication to be set on payment.
         '''
-        return ' '.join(label for label in batch_result['lines'].mapped('name') if label)
+        name = ' '.join(label for label in batch_result['lines'].mapped('name') if label)
+        if not name.strip() and batch_result['lines']:
+            name = " ".join(move_id.payment_reference or move_id.ref or move_id.name for move_id in batch_result['lines'].mapped('move_id'))
+        return name
```

Comment:
No way... The payment wizard should only be based on line's name to work correctly with matching rules.